iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -67,6 +81,20 @@ PoshRuntime& PoshRuntime::getInstance(cxx::optional<const RuntimeName_t*> name)
     return getRuntimeFactory()(name);
 }
 
+cxx::ScopeGuard PoshRuntime::getLifetimeParticipant() noexcept
+{
+    return cxx::ScopeGuard(
+        [](){
+            ++refcount;
+        },
+        [](){
+            if (0 == --refcount && manual_lifetime_management) {
+                getInstance().~PoshRuntime();
```

Comment:
I'm wondering if we could get rid of `manual_lifetime_management` but I guess we need it.

It also took me some time to understand why exactly it is needed. If I understood it correctly, this is for cases where `getLifetimeParticipant` is called before `initRuntime`. In this case we cannot rely on the refcount being 0 and return an empty scope guard but must cover the case the `initRuntime` is called later on.
Could you maybe add a comment to make it more clear?

Reply:
We do rely on the refcount being always 0 before everything else since it is a load-time constant.

Yes, this protects against the lifetime participant going out of scope before the factory was ever called, but primarily the purpose is for child classes like `PoshRuntimeSingleProcess`, which live in a regular static variable instead of a manually-managed buffer, and where we need to prevent the destructor from being called twice (by the scope guard and the static destruction).

Reply:
... oh, you could also wrap it into a anonymous namespace then it should not be accessible from other translation units.

Reply:
I think it's not necessary: https://stackoverflow.com/a/5622848/1726797 So I'd leave it like it is, but add a comment, ok?

Reply:
@nnmm well, that is for an integer and the solution was to not initialize it but atomics have some weird/unintuitive initialization behavior and there is always Axivion which complains about not initialized variables. We also already had initialization issues with the runtime so the chicken in me leans towards the approach with the function :sweat_smile:

Reply:
Ah okay, I didn't know that atomics have weird initialization behavior (do you have a link for that?). I applied your proposal.

Reply:
Oh I heared c++ atomics and weird initialization behavior, just look here: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p0883r2.pdf